apis_core/apis_entities/filters.py

```diff
@@ -84,16 +85,54 @@ def set_sort_filters(self, default_filters):
         :return: ordered dictionary with tuples of field names
                  and filters defined for them
         """
-        field_filters = OrderedDict()
+        sorted_filters = OrderedDict()
 
-        fields = []
+        list_filters = []
         if settings.APIS_ENTITIES:
             entity_settings = settings.APIS_ENTITIES.get(self.entity_class.__name__, {})
-            fields = entity_settings.get("list_filters", [])
-            if fields == ["name", "related_entity_name", "related_property_name"]:
-                # for AbstractEntity, certain fields are defined to be filterable by default;
-                # ignore these for the purpose of sorting field names for filtering
-                fields = []
+            list_filters = entity_settings.get("list_filters", [])
+
+        # This is a bit of a handful, but basically we have a list of strings and dicts
+        # and we use this list comprehension to create tuples of (filtername, None) for the
+        # strings and tuples of (filtername, values) for the dicts
+        list_filters = [
+            (f, None)
+            if isinstance(f, str)
+            else (list(f.keys())[0], f[list(f.keys())[0]])
+            for f in list_filters
+        ]
+
+        # The `list_filters` setting of an entity defines both the order of filters and it allows to define additional filters
+        # We differentiate three types of filter settings
+        # * a string: the name of a default filter of that model (simply to put it at the right position)
+        # * a dict with the name of a default filter as key and a dict of additional attributes as value:
+        #   to put it at the right position *and* set the attributes
+        # * a dict with a key that is *not* a name of a default filter with a dict of attributes:
+        #   to create a new filter. The attribute dict has to contain the `field_name` key - the new
+        #   filter will be based on the filter for that field
+        for filtername, filter_definition in list_filters:
+            if filtername in filters:
+                sorted_filters[filtername] = filters.pop(filtername)
+                if isinstance(filter_definition, dict):
+                    for key, value in filter_definition.items():
+                        setattr(sorted_filters[filtername], key, value)
+            elif (
+                isinstance(filter_definition, dict)
+                and "field_name" in filter_definition.keys()
+            ):
+                field_name = filter_definition["field_name"]
+                sorted_filters[filtername] = copy.copy(filters[field_name])
+                for key, value in filter_definition.items():
+                    setattr(sorted_filters[filtername], key, value)
+
+        unused_filters = [x for x in list_filters if x[0] not in sorted_filters.keys()]
```

Comment:
Not sure what you mean exactly - if I've defined a filter two times, the second one is the one that gets used

Reply:
(Side note: I'll have to watch if I misclick the targeted lines while reviewing or if GitHub does something weird with in-line comments – like in another recent PR, I was meaning to highlight a line a few lines below the one that gets marked as first here, i.e. 128 instead of 125.)

I just realised I was basing this on the incorrect premise that the validity of a _filter definition_ would be checked for as well, in which case I would have expected incorrect definitions to get added to `unused_filters`.

Example:
```
            "known_field",
            {"known_field": {"hello": "world"}},
            {"known_field": {"method": "what"}},
```
-> The assumption was the first one would get added to `sorted_filters` and the latter two would get ignored. `unused_filters = [x for x in list_filters if x[0] not in sorted_filters.keys()]` would not have caught them because it would have found a match for `known_field` in `sorted_filters`.

I.e. my bad.